### PR TITLE
Basic consent implementation

### LINKED ISSUE #72: Basic implementation of consents

### PR DESCRIPTION
**Pull request by amundlrohne**
Consent entity, routes, service, and frontend integration. 

Missing:
- User administration. 
- Admin notifications

Closes #72

### CODE REVIEW

File: apps/api/src/services/UserService.ts [hunk 1]
```diff
@@ -48,6 +48,10 @@ export class UserService extends BaseService<User> {
         const users = await this._userRepository
             .createQueryBuilder('user')
             .leftJoinAndSelect('user.criterias', 'criteria', 'criteria.id IN (:...criteriaIds)', { criteriaIds })
+            .groupBy('user.id')
+            .addGroupBy('criteria.id')
+            .having('COUNT(criteria.id) > 0')
+            // .orderBy('COUNT(criteria.id)', 'DESC') Why doesn't this work?
```
Comment (perkynades) [inline apps/api/src/services/UserService.ts:54]: lol

Comment (perkynades) [inline apps/api/src/services/UserService.ts:54]: lol

File: apps/frontend/src/app/components/insightConfiguration/ConsentsSearchModal.tsx [hunk 2]
```diff
@@ -10,37 +11,29 @@ interface IProps {
...
+    };
 
     return (
         <Modal open={open} onClose={close}>
             <Modal.Content className={style.wrapper}>
-                <SearchField
-                    label="Kriterier"
-                    description={<div>Søk etter samtykker her</div>}>
-                    <SearchField.Input />
+                <SearchField label="Kriterier" description={<div>Søk etter samtykker her</div>}>
```
Comment (perkynades) [inline apps/frontend/src/app/components/insightConfiguration/ConsentsSearchModal.tsx]: Unsure if this is for the creation of consents, or if searching for consents connected to a person. But if it is for consents connected to a person, i think it instead should just list all consents connected to a person.

Comment (perkynades) [inline apps/frontend/src/app/components/insightConfiguration/ConsentsSearchModal.tsx]: Unsure if this is for the creation of consents, or if searching for consents connected to a person. But if it is for consents connected to a person, i think it instead should just list all consents connected to a person.

File: apps/frontend/src/app/components/misc/apiHandler/APIHandler.tsx [hunk 3]
```diff
@@ -22,6 +22,9 @@ export const APIHandler = ({ loading, error }: IProps): ReactElement => {
             case 403:
                 element = <Heading size="medium">Du har ikke tilgang til denne siden.</Heading>;
                 break;
+            case 400:
+                element = <Heading size="medium">Sjekk at alle feltene er utfylt.</Heading>;
+                break;
```
Comment (perkynades) [inline apps/frontend/src/app/components/misc/apiHandler/APIHandler.tsx:27]: This might be redundant when validation is up and running :)

Comment (perkynades) [inline apps/frontend/src/app/components/misc/apiHandler/APIHandler.tsx:27]: This might be redundant when validation is up and running :)

**Review by perkynades — COMMENTED** [review]
See my comment on the ConsentSearchModal

File: apps/frontend/src/app/components/insightConfiguration/ConsentsSearchModal.tsx (hunk 2 above)
Comment (amundlrohne) [reply apps/frontend/src/app/components/insightConfiguration/ConsentsSearchModal.tsx]: This is for adding consents to a particular piece of insight work. Assuming there are ready-made templates in the db.

Comment (amundlrohne) [reply apps/frontend/src/app/components/insightConfiguration/ConsentsSearchModal.tsx]: It seems there is a typo here tho.

Comment (amundlrohne) [reply apps/frontend/src/app/components/insightConfiguration/ConsentsSearchModal.tsx]: Duplicate comment?

File: apps/frontend/src/app/components/misc/apiHandler/APIHandler.tsx (hunk 3 above)
Comment (amundlrohne) [reply apps/frontend/src/app/components/misc/apiHandler/APIHandler.tsx:27]: Good to know, will keep in place until then :)

Comment (amundlrohne) [reply apps/frontend/src/app/components/misc/apiHandler/APIHandler.tsx:27]: Duplicate?

**Review by perkynades — APPROVED** [review]
lgtm

**Pull request merged**